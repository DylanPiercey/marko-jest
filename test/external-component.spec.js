const path = require('path');
const { init, cleanup } = require('../');

const { render } = init(path.resolve(__dirname, './resources/external-component/index.marko'));

describe('external-component', () => {
  let renderResult;

  afterEach(cleanup);

  describe('on rendering', () => {
    beforeEach(async () => {
      renderResult = await render({});
    });

    it('should render correctly', () => {
      expect(renderResult.getNodes()).toMatchSnapshot();
    });
  });

  describe('on triggering click handler', () => {
    let component;

    beforeEach(async () => {
      renderResult = await render({});
      ({ component } = renderResult);

      component.el.click();
      component.update();
    });

    it('should update the element', () => {
      expect(renderResult.getNodes()).toMatchSnapshot();
    });

    it('should change the button label', () => {
      const buttonLabel = component.el.textContent;
      expect(buttonLabel).toEqual('DONE');
    });
  });
});
