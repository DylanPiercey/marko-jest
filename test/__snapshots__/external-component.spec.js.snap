// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`external-component on rendering should render correctly 1`] = `
Array [
  <ebay-button
    priority="primary"
  >
    Click me
  </ebay-button>,
]
`;

exports[`external-component on triggering click handler should update the element 1`] = `
Array [
  <ebay-button
    priority="primary"
  >
    DONE
  </ebay-button>,
]
`;
